pkg/cloudprovider/providers/gce/gce_healthchecks.go

```diff
@@ -28,12 +28,14 @@ import (
 	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
 	"k8s.io/api/core/v1"
 	utilversion "k8s.io/apimachinery/pkg/util/version"
-	"k8s.io/kubernetes/pkg/master/ports"
 )
 
 const (
-	nodesHealthCheckPath   = "/healthz"
-	lbNodesHealthCheckPort = ports.ProxyHealthzPort
+	nodesHealthCheckPath = "/healthz"
+	// NOTE: Please keep the following port in sync with ProxyHealthzPort in pkg/master/ports/ports.go
+	// ports.ProxyHealthzPort was not used here to avoid dependencies to k8s.io/kubernetes in the
+	// GCE cloud provider which is required as part of the out-of-tree cloud provider efforts.
+	lbNodesHealthCheckPort = 10256
```

Comment:
Not ideal to duplicate the port value there but it will unblock us from being able to move the GCE provider out-of-tree. I think it's worth the trade-off.

Reply:
Shouldn't the ports move to some common api-like repository? E.g. well-known constants. Maybe we can drive that, rather than break these depedencies...

Reply:
Ideally yes, but this seems like something that would go through API change reviews. There are also differing opinions on where those port values should live and I would prefer not to be blocked on this for too long. Can we add a TODO here and I will try to follow-up on putting the ports in a common api-like repo (maybe k8s.io/api/v1/well_known_constants.go`)?

Reply:
Also we should *not* be using the hard-coded port as it's override-able at runtime. So this constant won't qualify for that repo either

Reply:
+1 for adding a TODO
I believe we should have a default but can be over-riden. 
However switching it to a default seems a separate effort from Andrew's dependency break.